File: dotnet/Lexy/Lexy.Compiler/Compiler/CSharp/BuiltInFunctions/MonthsFunctionCall.cs
using Lexy.Compiler.Language.Expressions.Functions;
using Lexy.RunTime.RunTime;

namespace Lexy.Compiler.Compiler.CSharp.BuiltInFunctions
{
    internal class MonthsFunctionCall : EndStartDateFunctionCall
    {
        protected override string ClassName => nameof(BuiltInDateFunctions);
        protected override string MethodName => nameof(BuiltInDateFunctions.Months);

        public MonthsFunctionCall(MonthsFunction function) : base(function)
        {
        }
    }
}

File: typescipt/src/compiler/JavaScript/builtInFunctions/monthsFunctionCall.ts
import {EndStartDateFunctionCall} from "./endStartDateFunctionCall";
import {MonthsFunction} from "../../../language/expressions/functions/monthsFunction";

export class MonthsFunctionCall extends EndStartDateFunctionCall {
   protected override className = "BuiltInDateFunctions";
   protected override methodName = "months";

   constructor(functionNode: MonthsFunction) {
      super(functionNode);
   }
}
